mlrun/projects/pipelines.py

```diff
@@ -537,6 +469,7 @@ def get_run_status(
         pass
 
 
+# TODO: should we move this entire class to a version specific KFP module? Either that or keep dual-compatibility
 class _KFPRunner(_PipelineRunner):
```

Comment:
I dont think the entire class needs to move but rather the implementation
this unit should hold only mlrun information (sdk) and the impl should be used from the newly pipelines folder

Reply:
That's a good point.

It seems to me that the main KFP usage points here are the `new_pipe_metadata` usage (which is already a function on the new pipelines module) and the `Compiler()` call under the `save` method. 

Since `Compiler` is also available under KFP 2.0, this might be a case of code that works on both versions. If, later down the line, we find out that this is not the case, we should be able to isolate it on the pipelines module without much trouble.